samples/42.scaleout/src/main/java/com/microsoft/bot/sample/scaleout/BlobStore.java

```diff
@@ -0,0 +1,122 @@
+import com.microsoft.azure.storage.StorageException;
+import com.microsoft.azure.storage.blob.BlobRequestOptions;
+import com.microsoft.azure.storage.blob.CloudBlobClient;
+import com.microsoft.azure.storage.blob.CloudBlobContainer;
+import com.microsoft.azure.storage.blob.CloudBlockBlob;
+import com.microsoft.bot.schema.Pair;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.http.HttpStatus;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * An implementation of the ETag aware IStore interface against Azure Blob Storage.
+ */
+public class BlobStore implements Store {
+
+    private final CloudBlobContainer container;
+    private final ObjectMapper objectMapper = new ObjectMapper().findAndRegisterModules();
+
+    public BlobStore(String accountName, String accountKey, String containerName) throws URISyntaxException, StorageException {
+        if (StringUtils.isBlank(accountName)) {
+            throw new IllegalArgumentException("accountName cannot be null");
+        }
+        if (StringUtils.isBlank(accountKey)) {
+            throw new IllegalArgumentException("accountKey cannot be null");
+        }
+        if (StringUtils.isBlank(containerName)) {
+            throw new IllegalArgumentException("containerName cannot be null");
+        }
+
+        // Create storage credential from name and key
+        StorageCredentials storageCredentials = new StorageCredentialsAccountAndKey(accountName, accountKey);
+        // Create storage account
+        CloudStorageAccount cloudStorageAccount = new CloudStorageAccount(storageCredentials, true);
+        CloudBlobClient client = cloudStorageAccount.createCloudBlobClient();
+        container = client.getContainerReference(containerName);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public CompletableFuture<Pair<JsonNode, String>> load(String key) {
+        if (StringUtils.isBlank(key)) {
+            throw new IllegalArgumentException("key cannot be null");
+        }
+        try {
+            CloudBlockBlob blob = container.getBlockBlobReference(key);
+            String content = blob.downloadText();
+            JsonNode obj = objectMapper.readTree(content);
+            String etag = blob.getProperties().getEtag();
+            return CompletableFuture.completedFuture(new Pair<>(obj, etag));
+        } catch (URISyntaxException e) {
+            e.printStackTrace();
+        } catch (StorageException e) {
+            if (e.getHttpStatusCode() == HttpStatus.NOT_FOUND.value()) {
+                return CompletableFuture.completedFuture(new Pair<>(objectMapper.nullNode(), null));
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        } finally {
+            return CompletableFuture.completedFuture(new Pair<>(null, null));
```

Comment:
Just a question, this `finally` block cannot be removed? or is added due to the multiple `catch` statements?

Reply:
It would be great to remove it but yes... it was added due to the multiple `catch`, you can check to apply a better implementation!

Reply:
As I'm not quite sure of what I want to do I will leave it for the testing/fixing moment